server/gateway/tests/endpoint/status-test.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+var chai = require('chai');
+var ZSchema = require('z-schema');
+var customFormats = module.exports = function(zSchema) {
+  // Placeholder file for all custom-formats in known to swagger.json
+  // as found on
+  // https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#dataTypeFormat
+
+  var decimalPattern = /^\d{0,8}.?\d{0,4}[0]+$/;
+
+  /** Validates floating point as decimal / money (i.e: 12345678.123400..) */
+  zSchema.registerFormat('double', function(val) {
+    return !decimalPattern.test(val.toString());
+  });
+
+  /** Validates value is a 32bit integer */
+  zSchema.registerFormat('int32', function(val) {
+    // the 32bit shift (>>) truncates any bits beyond max of 32
+    return Number.isInteger(val) && ((val >> 0) === val);
+  });
+
+  zSchema.registerFormat('int64', function(val) {
+    return Number.isInteger(val);
+  });
+
+  zSchema.registerFormat('float', function(val) {
+    // better parsing for custom "float" format
+    if (Number.parseFloat(val)) {
+      return true;
+    } else {
+      return false;
+    }
+  });
+
+  zSchema.registerFormat('date', function(val) {
+    // should parse a a date
+    return !isNaN(Date.parse(val));
+  });
+
+  zSchema.registerFormat('dateTime', function(val) {
+    return !isNaN(Date.parse(val));
+  });
+
+  zSchema.registerFormat('password', function(val) {
+    // should parse as a string
+    return typeof val === 'string';
+  });
+};
+
+customFormats(ZSchema);
+
+var validator = new ZSchema({});
+var supertest = require('supertest');
+var api = supertest('http://localhost:4000'); // supertest init;
+
+chai.should();
+
+describe('/status', function() {
+  describe('get', function() {
+    it('should respond with 200 The server status will be...', function(done) {
+      /*eslint-disable*/
+      var schema = {
+        "type": "object",
+        "required": [
+          "up_time"
+        ],
+        "properties": {
+          "up_time": {
+            "type": "integer"
+          }
+        }
+      };
+
+      api.get('/v1/status')
+      .set('Content-Type', 'application/json')
+      .expect(200)
+      .end(function(err, res) {
+        if (err) return done(err);
+
+        validator.validate(res.body, schema).should.be.true;
+        done();
+      });
+    });
+
+    it('should respond with 405 Method not supported Error...', function(done) {
```

Comment:
405 tests are not necessary, swagger will handle these errors

Reply:
I know swagger is supposed to handle these errors but we should write the test to ensure swagger works as expected.